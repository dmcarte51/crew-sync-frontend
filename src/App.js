import { BrowserRouter, Route, Routes } from 'react-router-dom';
import './App.css';
import background from './CrewSync.svg';
import Calendar from './components/Calendar';
import D3Emp from './components/D3Emp';
import D3Test from './components/D3Test';
import EmpAvail from './components/EmpAvail';
import Landing from './components/Landing';
import Layout from './components/Layout';
import Login from './components/Login';
import Register from './components/Register';


function App() {
  // Your employee data
  const empData =  [
    { name: 'David', emp_id: 5, startTimes: ['08:00', '10:00', '12:00', '08:00', '10:00', '10:00', '10:00'], endTimes: ['16:00', '14:00', '18:00', '14:00', '16:00', '16:00', '14:00'] },
    { name: 'John', emp_id: 6, startTimes: ['10:00', '08:00', '10:00', '10:00', '08:00', '10:00', '08:00'], endTimes: ['14:00', '16:00', '16:00', '14:00', '14:00', '16:00', '16:00'] },
    { name: 'Paul', emp_id: 7, startTimes: ['10:00', '08:00', '08:00', '10:00', '10:00', '08:00', '10:00'], endTimes: ['16:00', '16:00', '14:00', '14:00', '16:00', '14:00', '16:00'] },
    { name: 'George', emp_id: 8, startTimes: ['12:00', '08:00', '08:00', '12:00', '10:00', '12:00', '10:00'], endTimes: ['18:00', '14:00', '14:00', '18:00', '14:00', '18:00', '14:00'] },
    { name: 'Ringo', emp_id: 9, startTimes: ['10:00', '08:00', '12:00', '08:00', '10:00', '08:00', '10:00'], endTimes: ['14:00', '16:00', '18:00', '14:00', '16:00', '14:00', '16:00'] },
    { name: 'Mick', emp_id: 10, startTimes: ['08:00', '10:00', '08:00', '12:00', '08:00', '10:00', '12:00'], endTimes: ['16:00', '14:00', '16:00', '18:00', '14:00', '16:00', '18:00'] },
    { name: 'Keith', emp_id: 11, startTimes: ['08:00', '12:00', '10:00', '08:00', '10:00', '10:00', '08:00'], endTimes: ['16:00', '18:00', '14:00', '16:00', '14:00', '14:00', '10:00'] },
    { name: 'Charlie', emp_id: 12, startTimes: ['08:00', '08:00', '12:00', '10:00', '08:00', '12:00', '10:00'], endTimes: ['16:00', '14:00', '18:00', '14:00', '16:00', '18:00', '14:00'] },
    { name: 'Ronnie', emp_id: 13, startTimes: ['08:00', '10:00', '10:00', '08:00', '12:00', '10:00', '08:00'], endTimes: ['16:00', '14:00', '16:00', '14:00', '18:00', '14:00', '10:00'] }
  ];

  function findByName(name) {
    return empData.find(employee => employee.name === name);
  }

  const dates = ['', '10/2/2023', '10/3/2023', '10/4/2023', '10/5/2023', '10/6/2023', '10/7/2023', '10/8/2023']
  // RANDOM WHITESPACE AT INDEX 1 IS NECESSARY
  // FOR THE CALENDAR TO DISPLAY CORRECTLY

  const dayOfWeek = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
  return (
    <BrowserRouter>
      <div
        className="App"
        style={{
          backgroundImage: `url(${background})`,
          backgroundPosition: 'center',
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
          width: '100vw',
          height: '100vh',
        }}
      >
        <Routes>
          <Route
            path="/"
            element={
              <Layout> {/* Wrap the content of each route with the Layout component */}
                <Landing />
              </Layout>
            }
          />
            <Route
            path="/reg"
            element={
              <Layout> {/* Wrap the content of each route with the Layout component */}
                <Register />
              </Layout>
            }
          />
          <Route
            path="/login"
            element={
              <Layout>
                <Login />
              </Layout>
            }
          />
          <Route
            path="/cal"
            element={
              <Layout>
                <Calendar empData={empData} dates={dates} />
              </Layout>
            }
          />
          {dayOfWeek.map((day, index) => (
            <Route key={index} path={`d3/${day}`} element={<D3Test empData={empData} index={index} />} />
          ))}
          {empData.map((employee) => (
            <Route key={employee.name} path={`d3/${employee.name}`} element={<D3Emp empData={findByName(employee.name)} />} />
          ))}
          <Route
            path="/empavail"
            element={
              <Layout>
                <EmpAvail empData={empData} />
              </Layout>
            }></Route>
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
