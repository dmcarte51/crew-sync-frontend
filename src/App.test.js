import { fireEvent, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import App from './App';

jest.mock('d3', () => ({
  // Mock implementations or objects go here
}));

jest.mock('axios');

// Smoke Test
test('renders without crashing', () => {
  render(<App />);
});
// In your test file or Jest setup file
jest.mock('openai', () => ({
  // Mock implementations or objects go here
}));

// Routing Tests
describe('routing tests', () => {
  test.each([
    { path: '/', componentText: 'Landing Page Text' },
    { path: '/reg', componentText: 'Register Page Text' },
    { path: '/del', componentText: 'Delete Page Text' },
    // ... other paths
  ])('renders $path page', ({ path, componentText }) => {
    const history = createMemoryHistory();
    history.push(path);
    render(
      <Router history={history}>
        <App />
      </Router>
    );
    expect(screen.getByText(new RegExp(componentText, 'i'))).toBeInTheDocument();
  });
});

// Integration Test Example
test('navigates to the calendar page when link is clicked', () => {
  const history = createMemoryHistory();
  render(
    <Router history={history}>
      <App />
    </Router>
  );
  fireEvent.click(screen.getByText(/calendar/i)); // Assuming "calendar" is the text on a nav link
  expect(screen.getByText(/calendar page/i)).toBeInTheDocument(); // Replace with actual text on your calendar page
});

// ...additional integration tests for form submissions, etc...

// API Mocking
// If you use API calls in your components, mock them here.

// Snapshot Test
test('matches the snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
