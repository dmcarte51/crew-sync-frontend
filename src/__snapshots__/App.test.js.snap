// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
    style="background-image: url(test-file-stub); background-position: center; background-size: cover; background-repeat: no-repeat; width: 100vw; height: 100vh;"
  >
    <div>
      <div
        class="navbar"
      >
        <a
          class="menu-bars"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </a>
      </div>
      <nav
        class="nav-menu"
      >
        <ul
          class="nav-menu-items"
        >
          <li
            class="navbar-toggle"
          >
            <a
              class="menu-bars"
              href="/"
            >
              <svg
                fill="currentColor"
                fill-rule="evenodd"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z"
                />
              </svg>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
                />
              </svg>
              <span>
                Home
              </span>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/cal"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M112 880c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V460H112v420zm768-696H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v176h800V216c0-17.7-14.3-32-32-32z"
                />
              </svg>
              <span>
                Calendar
              </span>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/reg"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 244c176.18 0 319 142.82 319 319v233a32 32 0 0 1-32 32H225a32 32 0 0 1-32-32V563c0-176.18 142.82-319 319-319zM484 68h56a8 8 0 0 1 8 8v96a8 8 0 0 1-8 8h-56a8 8 0 0 1-8-8V76a8 8 0 0 1 8-8zM177.25 191.66a8 8 0 0 1 11.32 0l67.88 67.88a8 8 0 0 1 0 11.31l-39.6 39.6a8 8 0 0 1-11.31 0l-67.88-67.88a8 8 0 0 1 0-11.31l39.6-39.6zm669.6 0l39.6 39.6a8 8 0 0 1 0 11.3l-67.88 67.9a8 8 0 0 1-11.32 0l-39.6-39.6a8 8 0 0 1 0-11.32l67.89-67.88a8 8 0 0 1 11.31 0zM192 892h640a32 32 0 0 1 32 32v24a8 8 0 0 1-8 8H168a8 8 0 0 1-8-8v-24a32 32 0 0 1 32-32zm148-317v253h64V575h-64z"
                />
              </svg>
              <span>
                Register
              </span>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/empavail"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 224c-79.6 0-144 64.4-144 144s64.4 144 144 144 144-64.4 144-144-64.4-144-144-144zm64 150.3c0 5.3-4.4 9.7-9.7 9.7h-60.6c-5.3 0-9.7-4.4-9.7-9.7v-76.6c0-5.3 4.4-9.7 9.7-9.7h12.6c5.3 0 9.7 4.4 9.7 9.7V352h38.3c5.3 0 9.7 4.4 9.7 9.7v12.6zM320 368c0-27.8 6.7-54.1 18.2-77.5-8-1.5-16.2-2.5-24.6-2.5h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h347.1c-45.3-31.9-75.1-84.5-75.1-144zm-96-112c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128z"
                />
              </svg>
              <span>
                Availability
              </span>
            </a>
          </li>
          <li
            class="nav-text"
          >
            <a
              href="/display"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 224c-79.6 0-144 64.4-144 144s64.4 144 144 144 144-64.4 144-144-64.4-144-144-144zm64 150.3c0 5.3-4.4 9.7-9.7 9.7h-60.6c-5.3 0-9.7-4.4-9.7-9.7v-76.6c0-5.3 4.4-9.7 9.7-9.7h12.6c5.3 0 9.7 4.4 9.7 9.7V352h38.3c5.3 0 9.7 4.4 9.7 9.7v12.6zM320 368c0-27.8 6.7-54.1 18.2-77.5-8-1.5-16.2-2.5-24.6-2.5h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h347.1c-45.3-31.9-75.1-84.5-75.1-144zm-96-112c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128z"
                />
              </svg>
              <span>
                DisplayUsers
              </span>
            </a>
          </li>
        </ul>
      </nav>
       
      <div>
        <div>
          <h1>
            Welcome to Our App
          </h1>
          <p>
            Please log in or sign up to continue.
          </p>
          <div>
            <div>
              <h1>
                Login
              </h1>
              <form>
                <input
                  name="username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
                <input
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <button
                  type="button"
                >
                  Login
                </button>
              </form>
            </div>
            <div>
              <h2>
                Register
              </h2>
              <form>
                <input
                  name="username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
                <input
                  name="email"
                  placeholder="Email"
                  type="email"
                  value=""
                />
                <input
                  minlength="8"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <input
                  minlength="8"
                  name="confirmPassword"
                  placeholder="Confirm Password"
                  type="password"
                  value=""
                />
                <button
                  type="submit"
                >
                  Register
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
